packages/tpetra/core/src/Tpetra_Details_Behavior.cpp

```diff
@@ -124,6 +244,50 @@ bool Behavior::verbose ()
                                                    verboseDefault ());
 }
 
+bool Behavior::debug (const char name[])
+{
+  static std::once_flag debugFlag_;
+  static bool debugInitialized_ = false;
+  static std::map<std::string, bool> namedDebugValues_;
```

Comment:
Putting initialization of the `static std::map` here, instead of inside the function to call via `std::call_once`, breaks reëntrancy of `Behavior::debug`.  Imagine if two different threads enter `Behavior::debug` concurrently.  They may both invoke the constructor of `std::map` concurrently, which could cause troubles (note that the default constructor doesn't promise not to allocate memory; for example, `std::vector` may reserve space for a small number of entries by default, if not given an initial number of entries).

Reply:
I see.  How would `namedDebugValues_` be declared then?  A pointer?